src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -30,21 +30,33 @@ internal class UserDefinedFunction : TexlFunction
     {
         private readonly bool _isImperative;
         private readonly IEnumerable<UDFArg> _args;
-        private TexlBinding _binding;
+        private readonly UserDefinedFunctionLibrary _library;
 
-        public override bool IsAsync => _binding?.IsAsync(UdfBody) ?? false;
+        public override bool IsAsync
```

Comment:
>IsAsync [](http://example.com/codeflow?start=29&length=7)

is there any danger in this value being able to mutate on us?

Reply:
we are not deriving IsAsync from Binding or library anymore.